src/brod_topic_subscriber.erl

```diff
@@ -0,0 +1,346 @@
+              end,
+      {AckRef, CbFun}
+    end,
+  NewPendings = Pendings ++ lists:map(MapFun, Messages),
+  NewState = State#state{pending_messages = NewPendings},
+  _ = send_lo_cmd(?LO_CMD_PROCESS_MESSAGE),
+  {noreply, NewState};
+handle_info(?LO_CMD_PROCESS_MESSAGE, State) ->
+  {ok, NewState} = maybe_process_message(State),
+  {noreply, NewState};
+handle_info(?LO_CMD_START_CONSUMER(ConsumerConfig, CommittedOffsets,
+                                   Partitions0),
+             #state{ client = Client
+                   , topic  = Topic
+                   } = State) ->
+  ok = brod:start_consumer(Client, Topic, ConsumerConfig),
+  {ok, PartitionsCount} = brod:get_partitions_count(Client, Topic),
+  AllPartitions = lists:seq(0, PartitionsCount - 1),
+  Partitions =
+    case Partitions0 of
+      all ->
+        AllPartitions;
+      L when is_list(L) ->
+        PS = lists:usort(L),
+        case lists:min(PS) >= 0 andalso lists:max(PS) < PartitionsCount of
+          true  -> PS;
+          false -> erlang:error({bad_partitions, Partitions0, PartitionsCount})
+        end
+    end,
+  Consumers =
+    lists:map(
+      fun(Partition) ->
+        AckedOffset = case lists:keyfind(Partition, 1, CommittedOffsets) of
+                        {Partition, Offset} -> Offset;
+                        false               -> ?undef
+                      end,
+        #consumer{ partition    = Partition
+                 , acked_offset = AckedOffset
+                 }
+      end, Partitions),
+  NewState = State#state{consumers = Consumers},
+  _ = send_lo_cmd(?LO_CMD_SUBSCRIBE_PARTITIONS),
+  {noreply, NewState};
+handle_info(?LO_CMD_SUBSCRIBE_PARTITIONS, State) ->
+  {ok, #state{} = NewState} = subscribe_partitions(State),
+  _ = send_lo_cmd(?LO_CMD_SUBSCRIBE_PARTITIONS, ?RESUBSCRIBE_DELAY),
+  {noreply, NewState};
+handle_info({'DOWN', _Mref, process, Pid, Reason},
+            #state{consumers = Consumers} = State) ->
+  case lists:keyfind(Pid, #consumer.consumer_pid, Consumers) of
+    #consumer{partition = Partition} = C ->
+      Consumer = C#consumer{ consumer_pid  = ?DOWN(Reason)
+                           , consumer_mref = ?undef
+                           },
+      NewConsumers = lists:keyreplace(Partition, #consumer.partition,
+                                      Consumers, Consumer),
+      NewState = State#state{consumers = NewConsumers},
+      {noreply, NewState};
+    false ->
+      {noreply, State}
+  end;
+handle_info(_Info, State) ->
+  {noreply, State}.
+
+handle_call(Call, _From, State) ->
+  {reply, {error, {unknown_call, Call}}, State}.
+
+handle_cast({ack, Partition, Offset}, State) ->
+  AckRef = {Partition, Offset},
+  {ok, NewState} = handle_ack(AckRef, State),
+  {noreply, NewState};
+handle_cast(stop, State) ->
+  {stop, normal, State};
+handle_cast(_Cast, State) ->
+  {noreply, State}.
+
+code_change(_OldVsn, State, _Extra) ->
+  {ok, State}.
+
+terminate(_Reason, #state{}) ->
+  ok.
```

Comment:
no, subscribers may under some other supervisor tree.
they may get restarted quite soon.

also, if subscriber is down, consumers won't do anything.
i.e. it won't cost much.